cypress/integration/onboarding/multi_team/existing_channel_name_spec.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/*eslint max-nested-callbacks: ["error", 3]*/
+
+describe('Channel', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    // * Verify new public or private channel cannot be created with existing public channel name:
+    it('M14635 Should not create new channel with existing public channel name', () => {
+        // 2. Find how many public channels there are and store as origPublicChannelLenth
+        cy.get('#sidebarChannelContainer').children().contains('PUBLIC CHANNELS').parent().parent().siblings().its('length').then(($origPublicChannelLength) => {
+            // * Verify new public channel cannot be created with existing public channel name:
+            // 3. Verify channel with name 'Town Square' exists
+            cy.get('#sidebarChannelContainer').should('contain', 'Town Square');
+
+            // 4. Click on '+' button for Public Channel
+            cy.get('#createPublicChannel').click();
+
+            // 5. Type 'Town Square' in the input field for new public channel
+            cy.get('#newPublicChannelName').type('Town Square');
+
+            //6. Click 'Create New Channel' button
+            cy.get('button[type=submit]').click();
+
+            // * User gets a message saying "A channel with that name already exists on the same team"
+            cy.get('p').contains('A channel with that name already exists on the same team');
```

Comment:
Same new ID here as,
```suggestion
            cy.get('#createChannelError').contains('A channel with that name already exists on the same team');
```